src/bcf/mod.rs

```diff
@@ -1022,13 +1022,15 @@ mod tests {
                     .expect("Missing tag")[0],
                 format!("string{}", i + 1).as_bytes()
             );
+            let fs1_str_vec = record
+                .format(b"FS1")
+                .string()
+                .expect("Error reading string.");
+            assert_eq!(fs1_str_vec.len(), 2);
```

Comment:
This checks the string vector length is consistent with sample count. However I couldn't easily reproduce the the case I had where `n != ret`, as it happened in the middle of reading a large vcf file and the criteria is unknown.

Reply:
From what I understand from the docs you cited, `n` is really a watermark. So you probably had a record with two string entries before the record with one string entry where the problem occurred.

So maybe you can change your test to actually read in a `.vcf` file with two records. The first one with to string entries, the second record with only one string entry. I think that this should then trigger that behaviour.

Reply:
I think I know the issue. The test needs to reuse the shared buffer to trigger the behavior. I have updated the tests. Now they will fail on master and pass with this PR.